### PR TITLE
Add versions logging for build tools

### PR DESCRIPTION
**Pull request by azotcsit**
It would be nice to see what exact versions of tools are used for the build process. Related to the query in https://issues.apache.org/jira/browse/INFRA-22823.

### CODE REVIEW

File: build-scripts/cassandra-artifacts.sh [hunk 1]
```diff
@@ -18,6 +18,11 @@ command -v docker >/dev/null 2>&1 || { echo >&2 "docker needs to be installed";
 [ -f "build.xml" ] || { echo >&2 "build.xml must exist"; exit 1; }
 [ -d "${cassandra_builds_dir}" ] || { echo >&2 "cassandra-builds directory must exist"; exit 1; }
 
+# print debug information on versions
+ant -version
```
Comment (azotcsit) [inline build-scripts/cassandra-artifacts.sh:22]: Here are outputs from my local:
```
Apache Ant(TM) version 1.10.11 compiled on July 10 2021
pip 20.3.3 from /usr/local/lib/python3.8/dist-packages/pip (python 3.8)
virtualenv 20.0.17 from /usr/lib/python3/dist-packages/virtualenv/__init__.py
Docker version 20.10.11, build dea9396e18
```

File: build-scripts/cassandra-test.sh [hunk 2]
```diff
@@ -10,6 +10,10 @@ command -v ant >/dev/null 2>&1 || { echo >&2 "ant needs to be installed"; exit 1
 command -v git >/dev/null 2>&1 || { echo >&2 "git needs to be installed"; exit 1; }
 [ -f "build.xml" ] || { echo >&2 "build.xml must exist"; exit 1; }
 
+# print debug information on versions
```
Comment (azotcsit) [inline build-scripts/cassandra-test.sh:13]: Here are outputs from my local:
```
Apache Ant(TM) version 1.10.11 compiled on July 10 2021
git version 2.25.1
```

**Comment by michaelsembwever** [comment]
nice!

can we also please add
```
pip --version
virtualenv --version
``` 
into cassandra-dtest-pytest.sh

and 
```
docker --version
```
 into cassandra-dtest-pytest-docker.sh and cassandra-test-docker.sh

**Comment by azotcsit** [comment]
> pip --version
> virtualenv --version

Added pip3 (pip3 command is called instead pip), virtualenv and ant (it is also called).

> docker --version

Added to "start the docker container" part of the condition.

**Comment by azotcsit** [comment]
@michaelsembwever is there a way to trigger Jenkins build against this branch? Even though the changes seem to be trivial and safe, it would be nice to ensure they break nothing.

**Comment by michaelsembwever** [comment]
> @michaelsembwever is there a way to trigger Jenkins build against this branch? Even though the changes seem to be trivial and safe, it would be nice to ensure they break nothing.

unfortunately no. this stuff gets tested against a local jenkins installation. (or cloning jobs, and manually adjusting their configuration to use this branch of cassandra-builds, under the [patches tab](https://ci-cassandra.apache.org/view/patches/)).

(ideally i dream about using the jenkins operator to be able to write a script that fires a full ci-cassandra.a.o clone up in a k8s cluster…)

**Review by michaelsembwever — APPROVED** [review]

**Comment by azotcsit** [comment]
The changes seem to be working fine, you can check https://ci-cassandra.apache.org/view/Cassandra%204.1/job/Cassandra-trunk/935/ build for more details.

**Pull request merged**